### PR TITLE
Clean up docs

### PR DESCRIPTION
**Pull request by aboedo**
Cleans up a few doc links that weren't working correctly

### CODE REVIEW

File: Purchases/Error Handling/StoreKitError+Extensions.swift [hunk 1]
```diff
@@ -34,7 +34,7 @@ extension StoreKitError {
         case .unknown:
             /// See also https://github.com/RevenueCat/purchases-ios/issues/392
             /// `StoreKitError` doesn't conform to `CustomNSError` as of `iOS 15.2`
-            /// so we can't extract any additional information like we do on ``SKError.toPurchasesErrorCode``
+            /// so we can't extract any additional information like we do on `SKError.toPurchasesErrorCode`
```
Comment (aboedo) [inline Purchases/Error Handling/StoreKitError+Extensions.swift:37]: links to other frameworks don't work, sadly, even if it's our own extension

File: Purchases/Identity/CustomerInfoManager.swift [hunk 2]
```diff
@@ -177,7 +177,8 @@ class CustomerInfoManager {
 
     /// Observers keyed by a monotonically increasing identifier.
     /// This allows cancelling observations by deleting them from this dictionary.
-    /// These observers are used both for ``customerInfoStream`` and `PurchasesDelegate.purchases(_:receivedUpdated:)`.
+    /// These observers are used both for ``Purchases/customerInfoStream`` and
```
Comment (aboedo) [inline Purchases/Identity/CustomerInfoManager.swift:180]: from a different object, you need to reference it as \`\`Type/property\`\`

File: Purchases/Public/Package.swift [hunk 3]
```diff
@@ -74,12 +74,12 @@ private extension PackageType {
     @objc public let storeProduct: StoreProduct
     @objc public let offeringIdentifier: String
 
-    /// The price of this product using ``StoreProduct.priceFormatter``.
+    /// The price of this product using ``StoreProduct/priceFormatter``.
```
Comment (aboedo) [inline Purchases/Public/Package.swift:77]: dots don't work, only slashes

File: RevenueCat.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -930,6 +930,7 @@
 				F530E4FE275646EF001AF6BD /* MacDevice.swift */,
 				57EAE52A274332830060EB74 /* Obsoletions.swift */,
 				2DDA3E4624DB0B5400EDFE5B /* OperationDispatcher.swift */,
+				2DD9F4BD274EADC20031AE2C /* Purchases+async.swift */,
```
Comment (aboedo) [inline RevenueCat.xcodeproj/project.pbxproj:933]: this is unrelated, I'll revert this bit

**Review by NachoSoto — APPROVED** [review]

**Pull request merged**